docs/source/geopmexporter.1.rst

```diff
@@ -0,0 +1,194 @@
+   Sum of all samples divided by the number of samples since last Prometheus
+   server scrape
+
+std:
+   Sample standard deviation represented by the following sum over samples:
+
+       ``sqrt(sum((x_i - m) ^ 2) / (n - 1))``
+
+   where ``m`` is the mean, ``x_i`` are the samples, and ``n`` is the number of
+   samples
+
+
+Prometheus Summary Method
+-------------------------
+
+When the user specifies the ``--summary prometheus`` command line option the
+``Summary`` and ``Counter`` metrics built into Prometheus are used to aggregate
+samples over time.  In this way the batch of signals read from the GEOPM is used
+to update a ``Summary`` metric for GEOPM signals with variable behavior, or a
+``Counter`` metric for signals with monotone behavior.  Each metric provides
+slightly different statistics for the Prometheus server on the port opened by
+``geopmexporter``.  the ``Summary`` metric exports the **count** and **sum**
+statistics, where as the ``Counter`` metric exports the **count** and **total**
+statistics.  GEOPM calls into the Prometheus metrics through a Python interface.
+This means that the quartile metrics are not currently supported for ``Summary``
+metrics. See the Prometheus documentation for more information about the
+``Summary`` and ``Counter`` metrics and the statistics that they export to the
+Prometheus server.
+
+Using Prometheus to aggregate samples provides an easy method to compare against
+the GEOPM ``StatsCollector`` method for summary.  Using the ``--summary prometheus``
+mode provides a more course grained view of the metrics gathered, and also tends
+to have a higher system overhead.
+
+
+Systemd Service
+---------------
+
+One simple way to manage the GEOPM Prometheus exporter process is to create a
+systemd service.  To use ``geopmexporter`` with systemd create a systemd unit
+configuration file called ``geopmexporter.service`` with the following content:
+
+.. code-block:: ini
+
+   [Unit]
+   Description=Prometheus GEOPM Exporter
+   StartLimitIntervalSec=0
+   StartLimitBurst=3
+   Wants=geopm.service
+   After=geopm.service
+   After=network-online.target
+
+   [Service]
+   Environment=PYTHONUNBUFFERED=true
+   User=prometheus
+   Group=prometheus
+   Type=simple
+   ExecStart=/usr/bin/geopmexporter
+   SyslogIdentifier=geopmexporter
+   Restart=always
+   RestartSec=1
+   StartLimitInterval=0
+
+   [Install]
+   WantedBy=multi-user.target
+
+
+Execute the commands below to install this systemd service and enable it.  These
+commands create a user and group called ``prometheus`` which will run the
+service, the systemd unit file described above is installed, and then systemd is
+notified of the new service.
+
+.. code-block:: bash
+
+   sudo groupadd --system prometheus
+   sudo useradd -s /sbin/nologin --system -g prometheus prometheus
+   sudo cp geopmexporter.service /etc/systemd/system/
+   sudo systemctl deamon-reload
+   sudo systemctl enable geopmexporter
+   sudo systemctl start geopmexporter
```

Comment:
We could also cook up an example of how to do this without using sudo, i.e. a userspace systemd service.  Not sure what all a user would want in this space.